### PR TITLE
ch4/ucx: Configure embedded UCX optimizations with --enable-fast

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Add optimization flags to UCX embedded configure if the user supplies
--enable-fast. Flags are copied from contrib/configure-opt in the UCX
source. Note that this enables non-portable CPU optimizations.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -902,6 +902,7 @@ for option in $enable_fast ; do
     esac
 done
 IFS="$save_IFS"
+export enable_fast
```
Comment (hzhou) [inline configure.ac:905]: Why export?

**Review by hzhou — APPROVED** [review]
LGTM. Just one non-important question.

File: configure.ac (hunk 1 above)
Comment (raffenet) [reply configure.ac:905]: In my tests, this wasn't visible in the subconfigure.m4 without export. I can try again without and verify.

Comment (raffenet) [reply configure.ac:905]: Looks like this does work with the export. I'll remove it.

**Comment by raffenet** [comment]
test:mpich/ch4/ucx

**Comment by hzhou** [comment]
test:mpich/custom --enable-fast=all
netmod: ch4:ucx

**Comment by raffenet** [comment]
Custom test looks good:

> configure:49907: RUNNING CONFIGURE FOR ch4:ucx
configure:50006: ===== configuring modules/ucx =====
configure:50171: running /bin/sh ./configure --disable-option-checking '--prefix=/var/lib/jenkins-slave/workspace/mpich-review-custom/_inst' --disable-static --enable-embedded --with-java=no --disable-logging --disable-debug --disable-assertions --disable-params-check --enable-optimizations '--with-device=ch4:ucx' '--disable-ft-tests' '--disable-spawn' '--enable-fast=all' '--enable-large-tests' '--enable-collalgo-tests' '--disable-perftest' '--with-wrapper-dl-type=rpath' 'CC=gcc -std=gnu99' 'CXX=g++' 'FC=gfortran' --cache-file=/dev/null --srcdir=.
configure:50191: ===== done with modules/ucx configure =====

**Pull request merged**